### ISSUE TITLE
running model.predict_generator gives different results each time

### ISSUE THREAD

**Issue by AlejandroBaron** [keras-team/keras#14599]
I'm trying to use generators in my CNN training but for some reason.
However, when I try to run model.predict_evaluator(), each time I run it (I'm working in Jupyter Notebook), ¡it gives different results! Same data (stored in folder), same model (I just rerun the same cell)

Here is how i create my generators, in case the issue is here. I set batch_size=1 to avoid any overlapping of folds (though I don't believe that's the issue)
```
train_datagen = ImageDataGenerator(
        preprocessing_function=preprocess_input,
        horizontal_flip=True)

test_datagen = ImageDataGenerator(preprocessing_function=preprocess_input)
train_generator = train_datagen.flow_from_directory(
        'keras_data/train',
        batch_size=1,
        class_mode='categorical')
validation_generator = test_datagen.flow_from_directory(
        'keras_data/val',
        batch_size=1,
        class_mode='categorical')
```

Then I train my model.

This block works fine, every time I rerun it, it gives the same metrics
```
test_generator = test_datagen.flow_from_directory(
            'keras_data/test',
            batch_size = 1,
            class_mode='categorical')
loss, acc = model.evaluate(test_generator, verbose=1)
print(loss,acc)
```

However, when I  run this cell, it gives different results every time
```
ytest = test_generator.classes
yhat = np.argmax(model.predict_generator(test_generator),axis=1)

from sklearn.metrics import confusion_matrix

m = confusion_matrix(ytest,yhat)
print(m)
```

It doesn't make any sense! Any ideas on what's happening? (using predict instead of predict_generator presents the same issue)
EDIT: enabled shuffling was the problem! solved
